traffic_ops/traffic_ops_golang/profile/profiles.go

```diff
@@ -226,10 +228,10 @@ LEFT JOIN cdn c ON prof.cdn = c.id`
 	return query
 }
 
-func ReadParameters(tx *sqlx.Tx, parameters map[string]string, user *auth.CurrentUser, profile tc.ProfileNullable) ([]tc.ParameterNullable, error) {
+func ReadParameters(tx *sqlx.Tx, user *auth.CurrentUser, profileID *int) ([]tc.ParameterNullable, error) {
 	privLevel := user.PrivLevel
 	queryValues := make(map[string]interface{})
-	queryValues["profile_id"] = *profile.ID
+	queryValues["profile_id"] = profileID
```

Comment:
why did we change the type of this query argument from `int` to `*int`? If it's `nil`, the query will always return zero results, which just means that it's a waste of a database call, so callers shouldn't use `ReadParameters` in that way.

Reply:
call was change to use the same function: ReadParameters throughout the code.

Reply:
And why would it be nil? Both times before calling this function, I am checking for profileID

Reply:
It shouldn't ever be `nil`, that's my point. If it can't ever be `nil` then there's no need for it to be a pointer.